### PR TITLE
vkd3d: Check for query pool when deduplicating query ranges.

### PR DESCRIPTION
**Pull request by doitsujin**

### CODE REVIEW

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -5043,10 +5043,10 @@ static void d3d12_command_list_read_query_range(struct d3d12_command_list *list,
      * in at most one range. */
     while (lo < hi)
     {
-        if (pos < list->query_ranges_count)
-        {
-            range = &list->query_ranges[pos];
+        range = &list->query_ranges[pos];
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:5046]: Dereferencing out of bounds like this seems questionable to me. I've had issues in the past where compiler assumed things based on that.

Comment (doitsujin) [reply libs/vkd3d/command.c:5046]: Shouldn't this be equivalent to `list->query_ranges + pos`? Guess we can write it like that explicitly but the actual dereference won't happen until after the range check anyways.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

File: libs/vkd3d/command.c (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d/command.c]: technically, it'd be a dereference, then addrof, and I think compiler is allowed to assume a dereferenced pointer to point to valid memory, so reorder could happen, but not 100% sure here, but the proposed fix should be fine.

**Pull request merged**